packages/oats-runtime/src/make.ts

```diff
@@ -111,12 +108,16 @@ export class Make<V> {
 
 export interface MakeOptions {
   unknownField?: 'drop' | 'fail';
+  parseNumberStrings?: boolean;
```

Comment:
these seem much nicer approach. I'm wondering if we would want to have a parsers for any value here instead of just for numbers and strings. Going with this might be ok but I'm kinda worried that we end up with an api that we want to change later leading to breakage. Maybe I worry too much though 🤷 

@ganymedes01 wdyt?

Reply:
the only place where we want to have parsing is parameters (query/path) which are strings. We don't want to have parsing for body, bc you can have any type you want there. Now what we want to parse besides numbers and booleans? Idt nulls or undefineds are good ideas.

Reply:
mmm. yes minimum amount of magic is preferable. Maybe we are fine with just numbers and strings. If not we can just add more options

Reply:
We might want to parse arrays also, which don't even have a standard implementation (or do but no one follows it). Currently we define it as a string and then `split(',')`

Reply:
i'll look into this. This pr is still WIP bc i forgot to reflect its changes in the client.

Reply:
I agree that this is a much better solution than the one before 👍 But I also think having something that works for any values (as sakari suggested) might be better, otherwise we could end up having to grow this list of options every time we find some exception

Reply:
Afaiu open api primitive data types are basically JSON data types except `null`.
So these are - `number`, `boolean`, `string`. And we got it all covered i think.

Reply:
but whos to say that query parameters will only be primitives?

Reply:
>We might want to parse arrays also, which don't even have a standard implementation (or do but no one follows it). Currently we define it as a string and then split(',')

i was thinking why do we need this if it's possible to provide an array in query parameters?
koa (and others like express) are automatically creating an array if query param repeats itself.
What we can do in this case is too always convert the string (or number/boolean) to an array if it consists only of one element.
e.g. `?someItems=Item1&someItems=Item2` => will be handled by koa: `someItems: ['Item1', 'Item2']`
`?someItems=Item0` => not handled by koa: `someItems: 'Item0'`

Reply:
but now i'm facing this really weird issue.
Axios is appending `[]` to query param if it's an array (wtf 😅).
Now this complicates the matter. If anyone knows how to disable this, i'd be really grateful.

Reply:
having repeated parameters suffixed with [] is one of the many ways to represent arrays in query parameters

Reply:
yes, but it's a convention, not a requirement. For some reason it's being forced when you use axios.
https://github.com/axios/axios/blob/master/lib/helpers/buildURL.js#L42
So if we want to support array schema for query parameters, the parameter names must be suffixed with `[]` (in openai spec), bc axios does it automatically (whether we want this or not). Now there's a way to provide `paramsSerializer`, but if we want to do we have to implement it.

Reply:
how the query parameters are mapped from and to a non scalar input is defined by the openapi spec. If we really want to we can work to support that bit better but that seems like lots of work and careful reading of the spec would be required as well as dealing with the various weird ways the adapters behave.

I would not do that now as I assume you have bunch of other stuff to do also.

Reply:
> I would not do that now as I assume you have bunch of other stuff to do also
yeah i've already done it. It shouldn't break much if axios is used to make requests to server that uses oats. Since currently making array schemas for query params work with oats is quite complicated, it's unlikely anyone uses it. But i can check in our services.
Now it should be working quite nicely.

Reply:
i can move it to another pr of course. It'll mean 2 major versions in a row (most likely), but i guess it's fine.

Reply:
mmm. people have reservations on doing major version updates. This might conceivably break cases where there is an oneOf  `array | string` as the type to work around this limitation of array handling (or number handling for that matter).

Reply:
i'll check how `array | string` will work. I'd assume, first come - first served. So if array is first (`array | string`), it'll always convert to an array for a single element. If it's a string first (`string | array`), nothing should change.

Reply:
Would we be ok by making this opt in as in https://github.com/smartlyio/oats/pull/246